### DISCUSSION TITLE
Does Model.count() accept an alias ?

### DISCUSSION THREAD

**Discussion by uzair004** [sequelize/sequelize#14292] [Help & Questions] [answered]
does Model.count() accept an alias to return count of the field by that alias instead of word count.

`let ordersCount = await Orders.count({
        where: { status: statuses },
        distinct: true,
        group: "status",
      });
`

I am passing an array of strings `statuses: ["statusA", "statusB",....]` & want to count orders based on provided statuses.

### Problem
this is what it returns 
`[
    {
         count: 10
     },
     {
         count: 20
     }
]`

### What I expect
`[
    {
         statusA: 10
     },
     {
         statusB: 20
     }
]`

I am using some logic to map through returned array but the problem is **Model.count doesn't return 0 count for status for which orders can't be found**

### What I want
either return count with dynamic provided name (alias)
or
return count: 0, where there are no orders for specific status

**Comment by ephys** [comment, marked as answer]
It looks like there is a bug. From what I see in the source code,

```typescript
const ordersCount = await Orders.count({ 
  where: { status: statuses }, 
  distinct: true, 
  group: "status", 
}); 
```

is supposed to return this:

```typescript
[{ status: 'statusA', count: 10 }, { status: 'statusB', count: 20 }]
```

You can work around it by setting the `attributes` option:

```typescript
const ordersCount = await Orders.count({ 
  attributes: ['status'],
  where: { status: statuses }, 
  distinct: true, 
  group: "status", 
}); 
```

(I don't think distinct is necessary here)

**Reply by uzair004** [reply]
removing `distinct: true` changes the order of result for some reasons.

also, sequelize skip those statuses where the count is zero, I was expecting count: 0 there, But that shouldn't be hard to solve :)

**Reply by uzair004** [reply]
Thanks for your answer 👍

**Reply by ephys** [reply]
> also, sequelize skip those statuses where the count is zero, I was expecting count: 0 there, But that shouldn't be hard to solve :)

That's due to how SQL works, SQL can't group by a value if it doesn't exist in the database
It's easy enough to handle in user code

> removing `distinct: true` changes the order of result for some reasons.

If you want a stable order you should use the `order` option (`order: 'status'`), it's still possible for the order of result to not be stable with `distinct` I think

**Reply by uzair004** [reply]
agree, but that would change it to alphabetical order while I want to get in a different order, the way statuses are written in the array. (not alphabetical order)
